Docker support for Hygieia

tyrion70 opened the pull request:
A simple setup to run Hygieia in a Docker container

srid99 commented:
@tyrion70 for UI setup, instead of using 'gulp serve' can you change it to use apache server ?

srid99 commented:
Gave a try by running `createdocker` script. The build went well but it failed when it tried `docker rm hygieia`.

```
Step 39 : CMD /usr/bin/supervisord -n
 ---> Running in 742586701a53
 ---> e970f0d0a90b
Removing intermediate container 742586701a53
Successfully built e970f0d0a90b
Error response from daemon: no such id: hygieia
FATA[0000] Error: failed to remove one or more containers 
```

tyrion70 commented:
Hmm that kinda makes sense. The first time u run it there is nog container yet. I'll fix that tonight thanks

tyrion70 commented:
Updated the createdocker script so it won't give an error the first time (it did start the container eventually right @srid99).
Also added Apache support, please anyone take a good look at what I did, I'm not sure if I have the document root completely correct. Its currently set to the /UI/dist folder now. I tried the src folder, but then it needed a symlink to include bower_components and it was missing css.
A Reverse proxy for /api was added to the apache configuration as well.

srid99 commented:
@tyrion70 Correct. It had error but the build was success and the container was running.

For the Apache support, what you did is correct. `dist` folder and the reverse proxy should do the job.

tabladrum commented:
@tyrion70: can you submit the CLA? Thanks

tyrion70 commented:
already did yesterday :) u didn't get it?

jdamick commented on Docker/Dockerfile:
```diff
@@ -0,0 +1,111 @@
+
+FROM ubuntu:14.04
+
+MAINTAINER Peter van Mourik <tyrion70@gmail.com>
```
should change this over to the group: Hygieia@capitalone.com

jdamick commented:
So in looking at this vs. pull #115 - to be honest i'd like to see something in the middle.. This pull has jenkins,sonar, etc all in the same container and lacks configurability.  The other pull has them separated out but lacks the collectors.. So I'm thinking about merging the best of both of them into a branch and then once it's solidified a bit merge it back to master.  Sound fair?

tyrion70 commented:
Hey! 

It's actually its different containers. One container is for the complete Hygieia solution including mongodb and apache. It also allows you to start other containers for sonar and jenkins (and mysql) using compose. 

Nevertheless there are some nice things in the other PR as well so feel free to solidify and merge it together :)

What do you mean by lack of configurability by the way? All config files needed are in the docker folder and can be modified before building the container. 

Let me know if you need any help merging or have any questions about what I did..

pjaol commented:
I've started down the path of dockerizing some of the collectors, will add those as separate pulls

jdamick commented:
closed by pull #129
